### ISSUE TITLE
Update Doxyfile to match the current sources

### ISSUE THREAD

**Issue by mouse07410** [open-quantum-safe/liboqs#820]
In short:
```diff
diff --git a/docs/.Doxyfile b/docs/.Doxyfile
index d162651a..b42ee8bf 100644
--- a/docs/.Doxyfile
+++ b/docs/.Doxyfile
@@ -792,9 +792,9 @@ WARN_LOGFILE           =
 # Note: If this tag is empty the current directory is searched.
 
 INPUT                  = src/common/common.h \
-                         src/common/rand.h \
-                         src/crypto/sha2/sha2.h \
-                         src/crypto/sha3/sha3.h \
+                         src/common/rand/rand.h \
+                         src/common/sha2/sha2.h \
+                         src/common/sha3/sha3.h \
                          src/kem/kem.h \
                          src/sig/sig.h \
                          README.md
@@ -2139,7 +2139,7 @@ TAGFILES               =
 # tag file that is based on the input files it reads. See section "Linking to
 # external documentation" for more information about the usage of tag files.
 
-GENERATE_TAGFILE       = NO
+#GENERATE_TAGFILE       = NO
 
 # If the ALLEXTERNALS tag is set to YES, all external class will be listed in
 # the class index. If set to NO, only the inherited external classes will be
```
Reason - `doxygen` v1.8.20 and newer require that `GENERATE_TAGFILE = filename`. It does not accept `NO`.

If you also do 
```
$ doxygen -u docs/.Doxyfile
```
it would be nice - but if not, users can do that on their own.

**Comment by dstebila** [comment]
Thanks. This was actually already being done as part of #817 but it's been held up by other changes, so I've pulled it out into a separate PR to be merged sooner.
